app/controllers/katello/api/v2/subscriptions_controller.rb

```diff
@@ -81,24 +81,18 @@ def create
       object = @system || @activation_key || @distributor
 
       if params[:subscriptions]
-        params[:subscriptions].each do |sub|
-          subscription = Pool.find(sub[:id])
-          object.subscribe(subscription.cp_id, subscription[:quantity])
+        params[:subscriptions].each do |sub_params|
+          subscription = Pool.find(sub_params[:id])
+          object.subscribe(subscription.cp_id, sub_params[:quantity])
+          subscription.import_data
```

Comment:
Since we are adding complexity, with an additional call out to Candlepin, I am thinking this should be dynflowized to better deal with potential errors. @jlsherrill would you agree?